src/VecSim/vec_sim_tiered_index.h

```diff
@@ -60,6 +63,9 @@ class VecSimTieredIndex : public VecSimIndexInterface {
     VecSimQueryResult_List topKQuery(const void *queryBlob, size_t k,
                                      VecSimQueryParams *queryParams) override;
 
+    // Return the current state of the global write mode (async/in-place).
+    static VecSimWriteMode getWriteMode() { return VecSimIndexInterface::asyncWriteMode; }
```

Comment:
why do we need this function in this class?

Reply:
Since the write mode is relevant for tiered index(es) only. Would you expect something otherwise?